### ISSUE TITLE
Ignore no-unused-var variables that are valid when using ts-check

### ISSUE THREAD

**Issue by muuvmuuv** [eslint/eslint#12683]
**Tell us about your environment**

Node version: v12.13.1
npm version: v6.13.3
Local ESLint version: v6.7.2
Global ESLint version: v6.7.2 (Currently used)

**What parser (default, Babel-ESLint, etc.) are you using?**

VS Code plugin eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "root": true,
  "plugins": [
    "prettier"
  ],
  "extends": [
    "eslint:recommended",
    "prettier"
  ],
  "parserOptions": {
    "ecmaVersion": 9,
    "sourceType": "module"
  },
  "env": {
    "node": true,
    "es6": true
  },
  "rules": {
    "prettier/prettier": "error"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// @ts-check

import { ReleaseType } from 'semver'

/** @type {Array.<ReleaseType>} */
const SEMVER_INCREMENTS = [
  'patch',
  'minor',
  'major',
  'prepatch',
  'preminor',
  'premajor',
  'prerelease',
]
```

```bash
eslint --config scripts/.eslintrc scripts/version/index.js
```

**What did you expect to happen?**

ESLint recognises that I have set a type so it is not "really" unused.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  11:27  error  'ReleaseType' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by kaicataldo** [comment]
Just a quick note that `VS Code plugin eslint` isn't a parser, and it looks like from the configuration that this is running the default parser. 

I don't think this is a good fit for core, but it does seem useful. Have you checked in with https://github.com/typescript-eslint/typescript-eslint to see if that project would add this feature?

**Comment by josephzidell** [comment]
I am facing a similar issue with the normal eslint parser and this (simplified) typescript code:

```ts
interface J {
	name: string;
}

var K: J[] = [];
```

**Comment by muuvmuuv** [comment]
Closing this in favor of referenced issue
